sdk/src/main/java/com/akkaserverless/javasdk/view/View.java

```diff
@@ -16,22 +16,64 @@
 
 package com.akkaserverless.javasdk.view;
 
-import com.akkaserverless.javasdk.impl.AkkaServerlessAnnotation;
+import com.akkaserverless.javasdk.impl.view.ViewUpdateEffectImpl;
 
-import java.lang.annotation.ElementType;
-import java.lang.annotation.Retention;
-import java.lang.annotation.RetentionPolicy;
-import java.lang.annotation.Target;
+import java.util.Optional;
 
-/**
- * Transformation of incoming messages and previous state to updated state can be implemented in a
- * class annotated with <code>@View</code>. It must be registered with {@link
- * com.akkaserverless.javasdk.AkkaServerless#registerView AkkaServerless.registerView}.
- *
- * <p>The view class should define methods corresponding to the service calls (rpc) for view
- * updates. Those methods are annoted with {@link UpdateHandler}.
- */
-@AkkaServerlessAnnotation
-@Target(ElementType.TYPE)
-@Retention(RetentionPolicy.RUNTIME)
-public @interface View {}
+/** @param <S> The type of the state for this view. */
+public abstract class View<S> {
+
+  private Optional<UpdateContext> updateContext = Optional.empty();
+
+  /**
+   * Additional context and metadata for an update handler.
+   *
+   * <p>It will throw an exception if accessed from constructor.
+   */
+  protected final UpdateContext updateContext() {
+    return updateContext.orElseThrow(
+        () ->
+            new IllegalStateException("UpdateContext is only available when handling an update."));
+  }
+
+  /** INTERNAL API */
+  public final void setUpdateContext(Optional<UpdateContext> context) {
```

Comment:
While you are at it, could you rename this to `_internalSetUpdateContext`. I did similar rename  in https://github.com/lightbend/akkaserverless-java-sdk/pull/255 but didn't touch this to avoid conflicts.

I'd like to "hide" it because it's public but internal.